src/shared/body_relations/base_body_relation.h

```diff
@@ -163,22 +169,27 @@ class BaseContactRelation : public SPHRelation
 
     virtual void resizeConfiguration() override;
 
-        void allocateContactConfiguration() {
-            for (const auto & body : contact_configuration_)
-                contact_configuration_device_.emplace_back(body.size(), execution::executionQueue.getQueue());
-        }
-
-        void copyContactConfigurationToDevice() {
-            for (int k = 0; k < contact_configuration_.size(); ++k)
-                for (int i = 0; i < contact_configuration_.at(k).size(); ++i)
-                    contact_configuration_device_.at(k).at(i) = contact_configuration_.at(k).at(i);
-        }
-
-        void copyContactConfigurationFromDevice() {
-            for (int k = 0; k < contact_configuration_.size(); ++k)
-                for (int i = 0; i < contact_configuration_.at(k).size(); ++i)
-                    contact_configuration_.at(k).at(i) = contact_configuration_device_.at(k).at(i);
-        }
-	};
+    bool isDeviceConfigurationAllocated() { return device_configuration_allocated_; }
+
+    void allocateContactConfiguration() {
+        for (const auto & body : contact_configuration_)
+            contact_configuration_device_.emplace_back(body.size(), execution::executionQueue.getQueue());
+        device_configuration_allocated_ = true;
+    }
```

Comment:
allocatedConrtactConfigurationOnDevice for DeviceallocatedConrtactConfiguration?

Reply:
> allocatedConrtactConfigurationOnDevice for DeviceallocatedConrtactConfiguration?
allocatedConrtactConfigurationOnDevice or DeviceallocatedConrtactConfiguration?

Reply:
I would say `allocateContactConfigurationOnDevice`, if it's not too lengthy

Reply:
Are you referring to the method's name or the name of variable `device_configuration_allocated_`?